cli/cliui/table.go

```diff
@@ -41,3 +44,174 @@ func FilterTableColumns(header table.Row, columns []string) []table.ColumnConfig
 	}
 	return columnConfigs
 }
+
+// DisplayTable renders a table as a string. The input argument must be a slice
+// of structs (panics otherwise, even for empty slices). At least one field in
+// the struct must have a `table:""` tag containing the name of the column in
+// the outputted table.
+//
+// Nested structs are processed if the field has the `table:"$NAME,recursive"`
+// tag and their fields will be named as `$PARENT_NAME $NAME`. If the tag is
+// malformed or a field is marked as recursive but does not contain a struct or
+// a pointer to a struct, this function will panic (even with an empty input
+// slice).
+//
+// If sort is empty, the input order will be used. If filterColumns is empty or
+// nil, all available columns are included.
+func DisplayTable(out interface{}, sort string, filterColumns []string) (string, error) {
+	v := reflect.Indirect(reflect.ValueOf(out))
+
+	if v.Kind() != reflect.Slice {
+		panic("DisplayTable called with a non-slice type")
+	}
+
+	headersRaw := typeToTableHeaders(v.Type().Elem())
+	if len(headersRaw) == 0 {
+		panic("no table tags found on the input type")
+	}
+	headers := make(table.Row, len(headersRaw))
+	for i, header := range headersRaw {
+		headers[i] = header
+	}
+
+	tw := Table()
+	tw.AppendHeader(headers)
+	tw.SetColumnConfigs(FilterTableColumns(headers, filterColumns))
+	tw.SortBy([]table.SortBy{{
+		Name: sort,
+	}})
+
+	// Write each struct to the table.
+	for i := 0; i < v.Len(); i++ {
+		// Format the row as a slice.
+		rowMap := valueToTableMap(v.Index(i))
+		rowSlice := make([]interface{}, len(headers))
```

Comment:
Suggestion (non-blocking): It's a post-generics crazy world, consider:

```suggestion
		rowSlice := make([]any, len(headers))
```

(There are more instances, feel free to reject this suggestion if you don't want to make the changes.)

Reply:
I don't mind if you want to change them, but if we care we should really add a linter for this.

Originally this function was gonna accept []T but I decided against it because I didn't want to clutter the binary with duplicated functions

Reply:
I don't have a huge preference, but I see we have mixed use and it'd be nice to standardize. As you propose, a linter would be a great idea.